packages/merge-styles/src/styleToClassName.ts

```diff
@@ -0,0 +1,179 @@
+import { GlobalSettings } from '@uifabric/utilities/lib/GlobalSettings';
+import { rtlifyRules } from './transforms/rtlifyRules';
+import { provideUnits } from './transforms/provideUnits';
+import { prefixRules } from './transforms/prefixRules';
+import { expandQuads } from './transforms/expandQuads';
+import { getVendorSettings } from './getVendorSettings';
+import { Stylesheet } from './Stylesheet';
+import { IStyle } from './IStyle';
+
+const DISPLAY_NAME = 'displayName';
+
+function toRuleEntry(name: string, value: string): string | undefined {
+  if (name === DISPLAY_NAME) {
+    return undefined;
+  }
+
+  // Prefixes like WebKit need a dash infront of them.
+  if (/[A-Z]/.test(name[0])) {
+    name = '-' + name;
+  }
+
+  // Kebab case the rule.
+  let rules: string[] = [
+    name.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase(),
+    value
+  ];
+
+  let ruleEntries: string[] = [];
+
+  // Apply transforms.
+  provideUnits(rules);
+  rtlifyRules(rules);
+  prefixRules(rules);
+  expandQuads(rules);
+
+  // Stringify the rules.
+  for (let i = 0; i < rules.length; i += 2) {
+    ruleEntries.push(rules[i], ':', rules[i + 1], ';');
+  }
+
+  return ruleEntries.join('');
+}
+
+function getDisplayName(rules?: Map<string, IStyle>): string | undefined {
+  let rootStyle: IStyle = rules && rules.get('&');
+
+  // tslint:disable-next-line:no-any
+  return rootStyle && (rootStyle as any).displayName;
+}
+
+/**
+ * Given an array of objects or strings, iterate through each:
+ *  if string, look up Map. Merge map into current map.
+ *  else if object:
+ *    iterate through each prop:
+ *      if is a selector.
+ *
+ *
+ * edge cases; should :active always have prescidence over :hover?
+ * ([':hover': {}, ':active': {}], [':hover': {}])
+ */
+function extractRules(
+  args: IStyle[],
+  rules: Map<string, {}> = new Map<string, {}>(),
+  currentSelector: string = '&'
+): Map<string, {}> {
+  let stylesheet = Stylesheet.getInstance();
```

Comment:
I would prefer it if most of these functions took a `Stylesheet` as a parameter rather than referring back to `Stylesheet.getInstance()`. It would produce a more decoupled architecture that would be easier to extend.